handlers/whatsapp/whatsapp.go

```diff
@@ -147,6 +219,36 @@ func (h *handler) receiveEvent(ctx context.Context, channel courier.Channel, w h
 	return events, courier.WriteDataResponse(ctx, w, http.StatusOK, "Events Handled", data)
 }
 
+func resolveMediaURL(channel courier.Channel, mediaID string) (string, error) {
+	token := channel.StringConfigForKey(courier.ConfigAuthToken, "")
+	if token == "" {
+		return "", fmt.Errorf("Missing token for WA channel")
+	}
+
+	urlStr := channel.StringConfigForKey(courier.ConfigBaseURL, "")
+	url, err := url.Parse(urlStr)
+	if err != nil {
+		return "", fmt.Errorf("Invalid base url set for WA channel: %s", err)
+	}
+
+	mediaPath, _ := url.Parse("/v1/media")
+	mediaEndpoint := url.ResolveReference(mediaPath).String()
+
+	fileURL := fmt.Sprintf("%s/%s", mediaEndpoint, mediaID)
+
+	return fileURL, nil
+}
+
+// BuildDownloadMediaRequest to download media for message attachment with Bearer token set
+func (h *handler) BuildDownloadMediaRequest(ctx context.Context, b courier.Backend, channel courier.Channel, attachmentURL string) (*http.Request, error) {
+	token := channel.StringConfigForKey(courier.ConfigAuthToken, "")
```

Comment:
What if the token is empty?

Reply:
Right now it'll fail on the `Authorization`, I can add a check on non-empty though.

Reply:
done